src/Core/Infrastructure/RealTime/Hypermedia/AbstractHypermediaProvider.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * Copyright 2005 - 2022 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\Infrastructure\RealTime\Hypermedia;
+
+use Centreon\Domain\Contact\Interfaces\ContactInterface;
+
+abstract class AbstractHypermediaProvider
+{
+    /**
+     * Checks if contact has access to pages defined in roles
+     *
+     * @param ContactInterface $contact
+     * @param string[] $topologyRoles
+     * @return boolean
+     */
+    protected function canContactAccessPages(ContactInterface $contact, array $topologyRoles): bool
+    {
+        if (! $contact->isAdmin() && ! $this->hasTopologyAccess($contact, $topologyRoles)) {
+            return false;
+        }
+
+        return true;
```

Comment:
avoid double negation
```suggestion
        return $contact->isAdmin() || $this->hasTopologyAccess($contact, $topologyRoles));
```